engine/inspector/private/inspector_module.cpp

```diff
@@ -0,0 +1,89 @@
+#include "inspector_module.hpp"
+#include "editor.hpp"
+#include "imgui_backend.hpp"
+#include "implot/implot.h"
+#include "menus/performance_tracker.hpp"
+#include "profile_macros.hpp"
+#include "renderer.hpp"
+#include "renderer_module.hpp"
+
+#include <scripting_module.hpp>
+
+InspectorModule::InspectorModule() = default;
+
+InspectorModule::~InspectorModule()
+{
+}
+
+ModuleTickOrder InspectorModule::Init(Engine& engine)
+{
+    ImGui::CreateContext();
+    ImPlot::CreateContext();
+
+    ImGuiIO& io = ImGui::GetIO();
+    (void)io;
+    io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard; // Enable Keyboard Controls
+    io.ConfigFlags |= ImGuiConfigFlags_NavEnableGamepad; // Enable Gamepad Controls
+
+    auto& ecs = engine.GetModule<ECSModule>();
+    auto& renderer = engine.GetModule<RendererModule>();
+    auto& window = engine.GetModule<ApplicationModule>();
+
+    _imguiBackend = std::make_shared<ImGuiBackend>(
+        renderer.GetRenderer()->GetContext(),
+        window, renderer.GetRenderer()->GetSwapChain(),
+        renderer.GetRenderer()->GetGBuffers());
+
+    _editor = std::make_unique<Editor>(ecs, renderer.GetRenderer(), _imguiBackend);
+    _performanceTracker = std::make_unique<PerformanceTracker>();
+
+    return ModuleTickOrder::ePostTick;
+}
+
+void InspectorModule::Shutdown(Engine& engine)
+{
+    if (auto* ptr = engine.GetModuleSafe<RendererModule>())
+    {
+        ptr->GetRenderer()->FlushCommands();
+    }
+    _editor.reset();
+}
+
+void InspectorModule::Tick(Engine& engine)
+{
+    _imguiBackend->NewFrame();
```

Comment:
i would move all the other editor logic to here as well probably

Reply:
I believe I got all of the logic, what am I missing something?

Reply:
well the editor draw is basically what draws all our imgui windows right, bit of an odd distinction to put the imgui titlebar in here but the rest of the editor logic in said function

Reply:
Oh yeah you are right... I just thought moving all the editor code outside of that class could take some time and make the inspector module. I will add a TODO here

Reply:
alright good enough, i also mentioned for the future since this is out of scope for this pr